core/domain/state_domain.py

```diff
@@ -5545,3 +5613,23 @@ def from_dict(
             state_version_history_dict['state_name_in_previous_version'],
             state_version_history_dict['committer_id']
         )
+
+
+UnionOfCustomizationArgsDictValues = Union[
```

Comment:
Why are these defined at the bottom of the file, rather than at the top?

Reply:
Because If we define this type at the top of the file then backend tests will fail with `undefined` error.

### this is wrong case

```python
UnionOfCustomizationArgsDictValues = Union[
    str,
    int,
    bool,
    List[str],
    List[SubtitledHtml],
    List[SubtitledHtmlDict],
    SubtitledHtmlDict,
    SubtitledUnicode, --- # Here this is referenced before definition. ( similar for other classes as well )
    SubtitledUnicodeDict,
    domain.ImageAndRegionDict,
    domain.GraphDict
]

class SubtitledUnicode:
    """Value object representing subtitled unicode.""" 

....
```

### this is ok case

```python

class SubtitledUnicode:
    """Value object representing subtitled unicode.""" 

    ....

UnionOfCustomizationArgsDictValues = Union[  ----------- # At this point every class is defined so no error will occur.
    str,
    int,
    bool,
    List[str],
    List[SubtitledHtml],
    List[SubtitledHtmlDict],
    SubtitledHtmlDict,
    SubtitledUnicode, 
    SubtitledUnicodeDict,
    domain.ImageAndRegionDict,
    domain.GraphDict
]
```

Reply:
Oh, interesting. This is unusual so please could you add a comment explaining that? Something like "This union type depends on XYZ so it has to be defined after XYZ is defined."

Also should this go in an "if MYPY" block? Not sure.

Reply:
> This is unusual so please could you add a comment explaining that? Something like "This union type depends on XYZ so it has to be defined after XYZ is defined."

Done

> Also should this go in an "if MYPY" block? Not sure

This Union type is imported into other modules also where it is used with the cast, so while running backend tests those modules will again throw an error.